mir_eval/multipitch.py

```diff
@@ -0,0 +1,383 @@
+'''
+The goal of multiple f0 (multipitch) estimation and tracking is to identify
+all of the active fundamental frequencies in each time frame in a complex music
+signal.
+
+Conventions
+-----------
+Multipitch estimates are represented by a time series, and a corresponding list
+of arrays of frequency estimates. Frequency estimates may have any number of
+frequency values, including (represented by an empty array). Time values are in
+units of seconds and frequency estimates are in units of Hz.
+
+Estimate time series should ideally be equal to reference time series, but if
+this is not the case, the estimate time series is resampled using a nearest
+neighbor interpolation to match the estimate. Time values in the estimate time
+series that are outside of the range of the reference time series are given
+null (empty array) frequencies.
+
+By default, a frequency is "correct" if it is within 0.5 semitones of a
+reference frequency. The metrics are based on those described in
+[#poliner2007]_ and [#bay2009]_.
+
+Metrics
+-------
+* :func:`mir_eval.multipitch.accuracy`: Precision, Recall, and Accuracy scores
+  based on the number of esimated frequencies which are sufficiently close to
+  the reference frequencies.
+
+* :func:`mir_eval.multipitch.error_score`: Substitution, Miss, False Alarm and
+  Total Error scores based on the number of esimated frequencies which are
+  sufficiently close to the reference frequencies.
+
+
+:references:
+    .. [#poliner2007] G. E. Poliner, and D. P. W. Ellis, "A Discriminative
+    Model for Polyphonic Piano Transription", EURASIP Journal on Advances in
+    Signal Processing, 2007(1):154-163, Jan. 2007.
+    .. [#bay2009] Bay, M., Ehmann, A. F., & Downie, J. S. (2009). Evaluation of
+    Multiple-F0 Estimation and Tracking Systems. In ISMIR (pp. 315-320).
+'''
+
+import numpy as np
+import collections
+from scipy.interpolate import interp1d
+from . import util
+import warnings
+
+
+MAX_TIME = 30000.  # The maximum allowable time stamp (seconds)
+MAX_FREQ = 5000.  # The maximum allowable frequency (Hz)
```

Comment:
5000 is the nearest "clean" frequency value above C8, which is the highest piano note. I think it's a reasonable "highest value", but it came from my brain, and not from any convention. I know in melody land a much lower value is often used.

We could use 20,000 Hz, but it makes it harder to catch "bad" frequency units like cents.